aws-kafkaconnect-customplugin/src/main/java/software/amazon/kafkaconnect/customplugin/UpdateHandler.java

```diff
@@ -0,0 +1,207 @@
+package software.amazon.kafkaconnect.customplugin;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.services.kafkaconnect.KafkaConnectClient;
+import software.amazon.awssdk.services.kafkaconnect.model.DescribeCustomPluginRequest;
+import software.amazon.awssdk.services.kafkaconnect.model.DescribeCustomPluginResponse;
+import software.amazon.awssdk.services.kafkaconnect.model.TagResourceRequest;
+import software.amazon.awssdk.services.kafkaconnect.model.UntagResourceRequest;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnNotUpdatableException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class UpdateHandler extends BaseHandlerStd {
+
+    private Logger logger;
+
+    private final Translator translator;
+    private final ExceptionTranslator exceptionTranslator;
+    private final ReadHandler readHandler;
+
+    public UpdateHandler() {
+        this(new ExceptionTranslator(), new Translator(), new ReadHandler());
+    }
+
+    /**
+     * This is constructor is used for unit testing.
+     *
+     * @param exceptionTranslator
+     * @param translator
+     * @param readHandler
+     */
+    UpdateHandler(
+        final ExceptionTranslator exceptionTranslator,
+        final Translator translator,
+        final ReadHandler readHandler) {
+        this.translator = translator;
+        this.exceptionTranslator = exceptionTranslator;
+        this.readHandler = readHandler;
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<KafkaConnectClient> proxyClient,
+        final Logger logger) {
+        this.logger = logger;
+
+        final ResourceModel desiredModel = request.getDesiredResourceState();
+        final ResourceModel previousModel = request.getPreviousResourceState();
+
+        return ProgressEvent.progress(desiredModel, callbackContext)
+            .then(
+                progress -> proxy
+                    .initiate(
+                        "AWS-KafkaConnect-CustomPlugin::Update::ValidateResourceExists",
+                        proxyClient,
+                        desiredModel,
+                        callbackContext)
+                    .translateToServiceRequest(translator::translateToReadRequest)
+                    .makeServiceCall(this::validateResourceExists)
```

Comment:
Same as the delete comment, the validateResource logic is likely need to be expanded and move to after the updateAPI call dispatched and handles the resource to be sure we are verifying against the real status of the resource and avoid any race conditions